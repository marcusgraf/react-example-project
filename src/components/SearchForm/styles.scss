.SearchForm {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  z-index: 99;

  input {
    width: 100%;
    background-color: rgba(0, 0, 0, 0.7);
    border: 2px solid rgba(255, 255, 255, 0.7);
    border-radius: 30px;
    text-transform: capitalize;
    margin: 1% auto;
    display: block;
    max-width: 300px;
    color: #fff;
    padding: 6px 12px;
  }

  input:focus {
    outline: none;
  }
}
