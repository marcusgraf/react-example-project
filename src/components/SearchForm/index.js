/* @flow */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scss';

type Props = { onSearchValueChange: PropTypes.func };

class SearchForm extends Component<Props> {
  constructor(props) {
    super(props);
    this.placeholder = 'Search for a movie';
  }

  handleInputChange = () => {
    const name = this._name.value;
    this.props.onSearchValueChange(name);
  };

  render() {
    return (
      <div className={styles.SearchForm}>
        <input type="text" ref={(input) => { this._name = input; }} onChange={this.handleInputChange} placeholder={this.placeholder} />
      </div>
    );
  }
}

export default SearchForm;
