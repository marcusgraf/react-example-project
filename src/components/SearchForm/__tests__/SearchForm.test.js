import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';

import SearchForm from '../index';

describe('<SearchForm />', () => {
  test('renders', () => {
    const tree = renderer.create(
      <StaticRouter context={{}}>
        <SearchForm />
      </StaticRouter>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
