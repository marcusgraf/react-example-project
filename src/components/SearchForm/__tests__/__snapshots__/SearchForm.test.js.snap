// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> renders 1`] = `
<div
  className="SearchForm"
>
  <input
    onChange={[Function]}
    placeholder="Search for a movie"
    type="text"
  />
</div>
`;
